worlds/tunic/__init__.py

```diff
@@ -387,6 +389,22 @@ def extend_hint_information(self, hint_data: Dict[int, Dict[int, str]]) -> None:
                 if hint_text:
                     hint_data[self.player][location.address] = hint_text
 
+    def get_real_location(self, location: Location) -> Location:
+        # if it's not in a group, it's not in an item link
+        if location.player not in self.multiworld.groups or not location.item:
+            return location
+
+        # can only find all instances of an item
+        possible_locations = self.multiworld.find_item_locations(location.item.name, self.player, True)
```

Comment:
Every call to `MultiWorld.find_item_locations()` iterates through every location in the multiworld. This seems like it could be rather poor for performance in the case of many items being item linked.

Reply:
There's 3 potential solutions here that I'm seeing:
1. Cache the "possible locations" in a dict or something
2. Change this entirely to check for multiple item instances at the same time
3. Find a better way to see where an item-linked item ended up that won't just give me the same item each time

I guess I'll probably just do number 1, even though it'll look really sloppy